lib/CGameInterface.h

```diff
@@ -76,6 +77,7 @@ class DLL_LINKAGE CBattleGameInterface : public IBattleEventsReceiver
 
 	virtual ~CBattleGameInterface() {};
 	virtual void initBattleInterface(std::shared_ptr<Environment> ENV, std::shared_ptr<CBattleCallback> CB){};
+	virtual void initBattleInterface(std::shared_ptr<Environment> ENV, std::shared_ptr<CBattleCallback> CB, AutocombatPreferences autocombatPreferences){};

```

Comment:
Default implementation should call the other overload

Reply:
It would hide missing implementation problem - implemented callback by battleAI - unless somebody has better idea new callback is intended to be used when autobattle is involved, but if somebody doesn't do it then AI assumes using everything is allowed. Wrongly using callback with autobattle preferences for non-autobattle cases will not break anything if we do not change default AutocombatPreferences values

Reply:
4 options
- you make initBattleInterface with 2 parameters non virtual and call 3-parameter one with preferences set all to true. All implementations only implement the one with 3 parameters.
- you make both methods pure virtual and each implementation must implement both
- you make initBattleInterface with 3 parameters to call the one with 2 parameters by default
- default implementation for 3 parameters throws exception clearly showing that this method is not supported

Reply:
Otherwise if I call initBattleInterface with 3 parameters on StupidAI - it will not initialize internal CB and ENV and so on

Reply:
see new commit - I implemented it in stupid AI (aka option 2)

Reply:
Ok